src/components/ToDoList/ToDoList.tsx

```diff
@@ -0,0 +1,229 @@
+import React from "react";
+import { useState } from "react";
+import Button from "../Button/Button";
+import "./ToDoList.css";
+import { ITodoList } from "../../App";
+
+let countId = 0;
+// const initialTodos = [
+//   { index: 1, title: "First task", isCompleted: false },
+//   { index: 2, title: "Second task", isCompleted: false },
+//   { index: 3, title: "Third task", isCompleted: false }
+// ]
+interface ITodos {
+  todos: ITodoList[];
+  setTodos: React.Dispatch<React.SetStateAction<ITodoList[]>>;
+}
+
+export default function ToDoList({ todos, setTodos }: ITodos) {
+  const [inputValue, setInputValue] = useState<string>("");
+  const [inputError, setInputError] = useState<boolean>(false);
+  const [showTextEror, setShowTextError] = useState<string>("");
+  const [disableStatusButton, setDisableStatusButton] = useState<boolean>(true);
+  const [filter, setFilter] = useState<string>("all");
+  const [editingItem, setEditingItem] = useState<number | undefined>(undefined);
+  const [inputValueEditing, setInputValueEditing] = useState<string>("");
+  const [inputEditError, setInputEditError] = useState<boolean>(false);
+
+  function addTask(): void {
+    if (inputValue.trim() !== "" && inputValue.length <= 150) {
+      setTodos([
+        ...todos,
+        { index: countId++, title: inputValue, isCompleted: false },
+      ]);
+      if (todos.length == 0) {
+        setDisableStatusButton(false);
+      }
+    } else {
+      setInputError(true);
+      if (inputValue.trim() == "") {
+        setShowTextError("*You must write something!");
+      } else {
+        setShowTextError("*Your ToDo is too long, please shorten it!");
+      }
+    }
+    setInputValue("");
+  }
+  function handleSubmitInputByEnter(
+    event: React.KeyboardEvent<HTMLInputElement>
+  ): void {
+    if (event.code === "Enter") {
+      addTask();
+    }
+  }
+  function handleChangeInput(event: React.ChangeEvent<HTMLInputElement>): void {
+    setInputError(false);
+    setShowTextError("");
+    setInputValue(event.target.value);
+  }
+  function handleChangeCheckbox(id: number): void {
+    setTodos(
+      todos.map((todo) => {
+        if (todo.index === id) {
+          todo.isCompleted = !todo.isCompleted;
+        }
+        return todo;
+      })
+    );
+  }
+
+  function handleDelete(id: number): void {
+    setTodos(todos.filter((item) => item.index !== id));
+    if (todos.length == 1) {
+      setDisableStatusButton(true);
+    }
+  }
+  function handleChangeInputEditing(
+    event: React.ChangeEvent<HTMLInputElement>
+  ): void {
+    setInputValueEditing(event.target.value);
+  }
+  function handleEditingItem(id: number, inputTitle: string): void {
+    setEditingItem(id);
+    setInputValueEditing(inputTitle);
+  }
+  function handleEditingSave(id: number): void {
+    if (inputValueEditing.trim() !== "" && inputValueEditing.length <= 150) {
```

Comment:
Do not use magic numbers. It will be better to move to separate files like `constants.ts` and name it line `MY_SUPER_MEGA_MAGIC_NUMBER`

Reply:
Corrected